app/modules/intelligence/agents/custom_agents/custom_agent_router.py

```
@@ -0,0 +1,119 @@
+        return await custom_agent_controller.delete_agent(agent_id, user["user_id"])
+    except ValueError as ve:
+        raise HTTPException(status_code=404, detail=str(ve))
+    except Exception as e:
+        logger.error(f"Error deleting custom agent: {str(e)}")
+        raise HTTPException(status_code=500, detail="Internal server error")
+
+@router.put("/{agent_id}", response_model=Agent)
+async def update_custom_agent(
+    agent_id: str,
+    request: AgentUpdate,
+    db: Session = Depends(get_db),
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)